newdle/api.py

```diff
@@ -595,3 +605,33 @@ def send_deletion_emails(args, code):
         },
     )
     return '', 204
+
+
+@api.route('/newdle/<code>/export')
+@use_kwargs(
+    {
+        'format': fields.String(required=True, validate=OneOf(('csv', 'xlsx'))),
+    },
+    location='query',
+)
+def export_participants(code, format):
+    newdle = Newdle.query.filter_by(code=code).first_or_404('Invalid code')
+    if newdle.creator_uid != g.user['uid']:
+        raise Forbidden
+
+    if format == 'csv':
+        buffer = export_answers_to_csv(newdle)
+    else:
+        buffer = export_answers_to_xlsx(newdle)
+
+    mimetype = {
+        'csv': 'text/csv',
+        'xlsx': 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    }[format]
```

Comment:
I'd either but both in a mapping (`{'csv': ('text/csv', export_anwers_to_csv), ...}`) or set the mimetype in the if/else above.